### DISCUSSION TITLE
I can't pass as a human on this site

### DISCUSSION THREAD

**Discussion by fernandovs4** [seleniumbase/SeleniumBase#2235] [General]
Hello everything is fine? I tried to access this site as I wanted to access it using automation. But still, with seleniumbase, he realized that I am not human.
my code was this:
from seleniumbase import Driver

driver = Driver(uc=True, headed=True, agent="desktop")
driver.get("https://login.globo.com/login/6668?url=https%3A%2F%2Fvalor.globo.com%2Fimpresso%2F20231101%2F")
driver.sleep(200)
driver.quit()

 I'm doing something wrong?

I'm grateful for the answer!

**Comment by mdmintz** [comment]
You have `agent="desktop"`, but `desktop` is not a valid user-agent string for a Chrome browser. You should remove that parameter entirely so that SeleniumBase automatically sets the user-agent to one that would be expected for a human using a web browser.

**Comment by fernandovs4** [comment]
thank you very much for answering me.
I removed the agent, but the website still shows that it is a robot

**Reply by mdmintz** [reply]
Loading the site in UC Mode is working for me. Are you referring to getting detected after trying to log in to the side after entering a username/password and then logging in? You'll need to use more methods for that.

Here's a list of special UC Mode `driver` methods:

```python
# "driver"-specific methods added by SeleniumBase for UC Mode: "--uc" / uc=True
driver.uc_open(url)
driver.uc_open_with_tab(url)
driver.uc_open_with_reconnect(url, reconnect_time=None)
driver.uc_click(selector)
```

Start by opening the page with `driver.uc_open_with_reconnect(url, reconnect_time=None)` using a long reconnect_time so that you have enough time to manually enter login credentials and sign in. If you don't get detected, it means UC Mode is good enough to avoid detection. Then you can use methods such as `driver.uc_click(selector)` to try avoiding detection when you click to log in. That's not guaranteed to work, but it has worked depending on the website. The sure-fire way to avoid detection is by not having to perform selenium actions at all when you open a website in UC Mode. The special methods were described in this video: https://www.youtube.com/watch?v=5dMFI3e85ig

**Reply by Id1n** [reply]
could you tell me how can i use switch iframe in uc mode ?

**Reply by OpsecGuy** [reply]
> could you tell me how can i use switch iframe in uc mode ? 

This is basically the same as you would do it without UC mode.

Using driver obj:
1. Define driver object.
2. Find a target frame and assign a variable to it. Then...
driver.frame.switch_to(YOUR_IFRME)
3. Do your stuff there using defined driver object or whatever else.
4. Switch back from your frame into default content!

**Reply by mdmintz** [reply]
Lots of iframe example tests here: [SeleniumBase/examples/iframe_tests.py](https://github.com/seleniumbase/SeleniumBase/blob/master/examples/iframe_tests.py)
